### PR TITLE
Story/rhoaieng 2982 experiments initial infrastructure

### PR DESCRIPTION
**Pull request by Gkrumbach07**
closes: https://issues.redhat.com/browse/RHOAIENG-2982

## Description

- added routing for experiments
- added empty states for active and archived
- added `disablePipelineExperiments` feature flag

Screenshot 2024-02-15 at 4 57 14 PM
Screenshot 2024-02-15 at 4 57 09 PM

## How Has This Been Tested?

- set the `disablePipelineExperiments` feature flag to false
- make sure you see the new nav options pop up. (the old pipelines layout and routes should disappear and be disabled as it uses a conflicting routing scheme)
- go to the only nav option under experiments
- test that the tabs change the url
- test changing projects

## Test Impact

added tests to check for empty states

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/pages/pipelines/GlobalPipelinesRoutes.tsx
Comment (Gkrumbach07) [inline frontend/src/pages/pipelines/GlobalPipelinesRoutes.tsx:118]: I am passing in the tab for each route... which gets sent through the stream of components to
a) set the active tab
b) render the correct empty state

**Comment by Gkrumbach07** [comment]
cc @kywalker-rh @yannnz 

UX for empty states of pipeline experiments active and archived

Mocks: [empty active](https://www.figma.com/proto/etbN0ccCcZR1qKCpJXQNAg/Pipeline-phase-2?page-id=615%3A34927&type=design&node-id=2005-153962&viewport=401%2C-96%2C0.02&t=hix7x325vWWGWrfX-1&scaling=scale-down-width&starting-point-node-id=2005%3A151595) & [empty archive](https://www.figma.com/proto/etbN0ccCcZR1qKCpJXQNAg/Pipeline-phase-2?page-id=615%3A34927&type=design&node-id=2005-153977&viewport=401%2C-96%2C0.02&t=hix7x325vWWGWrfX-1&scaling=scale-down-width&starting-point-node-id=2005%3A151595)

**Comment by Gkrumbach07** [comment]
/hold

wait until next week to merge

File: frontend/src/pages/pipelines/global/experiments/GlobalNoExperiments.tsx
Comment (jpuzz0) [inline frontend/src/pages/pipelines/global/experiments/GlobalNoExperiments.tsx:27]: "Active" is meant to be bold here according to the design.

File: frontend/src/pages/pipelines/global/experiments/GlobalExperiments.tsx
Comment (jpuzz0) [inline frontend/src/pages/pipelines/global/experiments/GlobalExperiments.tsx:16]: nit: I think either `pipelinesAPI` or `pipelinesApi` vs the casing here.

File: frontend/src/utilities/NavData.tsx
Comment (jpuzz0) [inline frontend/src/utilities/NavData.tsx:56]: nit: There are a few TODOs I'm seeing, and wondering if there should be JIRA links associated with them.

**Review by jpuzz0 — APPROVED** [review]
Tested this out. Looks good. Only minor comments. Going to approve since I don't consider them blocking, but boldening "Active" for the empty state message in Archived should probably happen before merging.

File: frontend/src/utilities/NavData.tsx
Comment (Gkrumbach07) [reply frontend/src/utilities/NavData.tsx:56]: hmm yeah. im not sure we have a jira for this. It might just be tracked by the artifacts and experiments epic. i can go in and make a jira to link here

File: frontend/src/pages/pipelines/global/experiments/GlobalNoExperiments.tsx
Comment (Gkrumbach07) [reply frontend/src/pages/pipelines/global/experiments/GlobalNoExperiments.tsx:27]: missed that good catch

File: frontend/src/pages/pipelines/GlobalPipelinesRoutes.tsx
Comment (andrewballantyne) [reply frontend/src/pages/pipelines/GlobalPipelinesRoutes.tsx:114]: That's fine for me -- runs currently use a query param -- but they could be migrated to use a route part.

**Pull request merged**
